includes/class-wcsatt-product.php

```diff
@@ -0,0 +1,856 @@
+<?php
+/**
+ * WCS_ATT_Product API
+ *
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All the Things
+ * @since    2.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * API for working with subscription-enabled product objects.
+ *
+ * @class    WCS_ATT_Product
+ * @version  2.0.0
+ */
+class WCS_ATT_Product {
+
+	public static function init() {
+
+		// Allow WCS to recognize any product as a subscription.
+		add_filter( 'woocommerce_is_subscription', array( __CLASS__, 'filter_is_subscription' ), 10, 3 );
+
+		// Delete object meta in use by the application layer.
+		add_action( 'woocommerce_before_product_object_save', array( __CLASS__, 'delete_reserved_meta' ) );
+
+		// Make product prices scheme-dependent.
+		WCS_ATT_Scheme_Prices::add_price_filters();
+		WCS_ATT_Scheme_Prices::add_price_html_filters();
```

Comment:
That makes sense, though I wonder if that means the `add_price_filters` and `add_price_html_filters` should actually be in this class too, and calling the corresponding `WCS_ATT_Scheme_Prices` method. That would provide a clean separation of concerns.